### PR TITLE
Feature/opo page

### PR DESCRIPTION
**Pull request by SammySteiner**

### CODE REVIEW

**Comment by SammySteiner** [comment]
[OPO Page Updates](https://trello.com/c/h2TCte5v/14-opo-page-updates)

File: gatsby-config.js [hunk 1]
```diff
@@ -399,6 +399,128 @@ module.exports = {
                       },
                     ],
                   },
+                  {
```
Comment (jlhogan) [inline gatsby-config.js:402]: Can I get a quick one liner that explains what these files/fields are defining ?

File: src/components/map/map.js [hunk 2]
```diff
@@ -86,60 +91,106 @@ export default function Map({
...
-                  : dsaGeoData?.childGeoJson?.features
-                ).map(f => ({
-                  ...f,
-                  properties: {
-                    ...f.properties,
-                    tier: opoDataMap[f.properties.opo].tier,
-                  },
-                }))}
+                key={state ? state + "opos" : opo ? opo + "opos" : "state opos"}
+                data={
```
Comment (jlhogan) [inline src/components/map/map.js:150]: I find this ternary + object expansion hard to make sense of; would like comment that explains what is being composed + why (i.e. what logic is used in conditionals and what's the end result)

File: src/components/opoTable/opoTable.js [hunk 3]
```diff
@@ -10,18 +11,40 @@ import Tier from "../tier/tier";
...
+          ...col,
+          Cell: props => (
+            <Link
+              to={`/opo/${opos.find(opo => opo.name === props.value)?.opo}`}
+            >
+              {props.value}
+            </Link>
+          ),
+        };
+      } else if (
```
Comment (jlhogan) [inline src/components/opoTable/opoTable.js:44]: don't need else if here because of the early return on line 34, right? Would prefer consecutive `if`s then (and below on 71)

File: src/utils/utils.js [hunk 4]
```diff
@@ -33,5 +39,19 @@ export const formatPercent = percent =>
...
+  });
+};
+
 export const formatStateName = ({ abbreviation, name }) =>
   `${name} (${abbreviation.toLocaleUpperCase()})`;
+
+export const formatOpoName = ({ name, opo }) =>
+  `${name} (${opo.toUpperCase()})`;
+
+export const formatOPORank = opoDate => 99;
```
Comment (jlhogan) [inline src/utils/utils.js:57]: what does this do? placeholder impl?

**Review by jlhogan — APPROVED** [review]
Overall seems reasonable to me, although it seems that a lot of redundant and non trivial conditional checking is now happening in map and opoTable depending on if the data being viewed is state or opo level data. I don't immediately have a suggestion for a different approach, but my initial reaction is that it seems like we're adding complexity -- there are so many places we check `if state then... else if opo then... else` that it is 1) hard to follow exactly what happens in the component and 2) intimidating to imagine making future changes and needing to ensure all these various conditional checks are accounted for. Happy to think more on this if others agree that it's worth further consideration
